packages/image_picker/image_picker/example/ios/RunnerTests/ImagePickerPluginTests.m

```diff
@@ -27,87 +28,147 @@ - (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker;
-  }
+  // UIImagePickerControllerSourceTypeCamera is supported
+  OCMStub(ClassMethod(
+              [_mockUIImagePicker isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]))
+      .andReturn(true);
+
+  // UIImagePickerControllerCameraDeviceRear is supported
+  OCMStub(ClassMethod(
+              [_mockUIImagePicker isCameraDeviceAvailable:UIImagePickerControllerCameraDeviceRear]))
+      .andReturn(true);
+
+  // AVAuthorizationStatusAuthorized is supported
+  OCMStub([_mockAVCaptureDevice authorizationStatusForMediaType:AVMediaTypeVideo]).andReturn(3);
+
+  // Run test
   FLTImagePickerPlugin *plugin = [FLTImagePickerPlugin new];
   FlutterMethodCall *call =
       [FlutterMethodCall methodCallWithMethodName:@"pickVideo"
                                         arguments:@{@"source" : @(0), @"cameraDevice" : @(0)}];
   [plugin handleMethodCall:call
                     result:^(id _Nullable r){
                     }];
+
   XCTAssertEqual([plugin getImagePickerController].cameraDevice,
                  UIImagePickerControllerCameraDeviceRear);
 }
 
-- (void)testPluginPickImageDeviceCancelClickMultipleTimes {
-  if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
-    return;
-  }
+- (void)testPluginPickVideoDeviceFront {
+  // UIImagePickerControllerSourceTypeCamera is supported
+  OCMStub(ClassMethod(
+              [_mockUIImagePicker isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]))
+      .andReturn(true);
+
+  // UIImagePickerControllerCameraDeviceFront is supported
+  OCMStub(ClassMethod([_mockUIImagePicker
+              isCameraDeviceAvailable:UIImagePickerControllerCameraDeviceFront]))
+      .andReturn(true);
+
+  // AVAuthorizationStatusAuthorized is supported
+  OCMStub([_mockAVCaptureDevice authorizationStatusForMediaType:AVMediaTypeVideo]).andReturn(3);
+
+  // Run test
   FLTImagePickerPlugin *plugin = [FLTImagePickerPlugin new];
   FlutterMethodCall *call =
-      [FlutterMethodCall methodCallWithMethodName:@"pickImage"
+      [FlutterMethodCall methodCallWithMethodName:@"pickVideo"
                                         arguments:@{@"source" : @(0), @"cameraDevice" : @(1)}];
   [plugin handleMethodCall:call
                     result:^(id _Nullable r){
                     }];
-  plugin.result = ^(id result) {
 
-  };
-  [plugin imagePickerControllerDidCancel:[plugin getImagePickerController]];
-  [plugin imagePickerControllerDidCancel:[plugin getImagePickerController]];
+  XCTAssertEqual([plugin getImagePickerController].cameraDevice,
+                 UIImagePickerControllerCameraDeviceFront);
 }
 
-- (void)testPluginPickVideoDeviceFront {
-  if (![UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+#pragma mark - Test camera devices, no op on simulators
+- (void)testPluginPickImageDeviceCancelClickMultipleTimes {
+  if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
```

Comment:
Why are you aborting the test if the camera *is* available?

Reply:
Hi Stuart,
this change made by [this commit](https://github.com/flutter/plugins/commit/25a585dfeef6d2dc224fd0dddee327372de63782#) and it already exists in master as well and got merged in my PR. Do you think we should remove it? Or ask what was this test about and try to fix it?

Reply:
@cyanglaz Was this a bug, or is this intentionally only supposed to run on simulator? That seems bizarre.